include/zephyr/drivers/video.h

```diff
@@ -118,6 +118,66 @@ struct video_buffer {
 	uint32_t timestamp;
 };
 
+/**
+ * @brief video_frmival_type enum
+ *
+ * Supported frame interval type of a video device.
+ */
+enum video_frmival_type {
+	VIDEO_FRMIVAL_TYPE_DISCRETE = 1,
+	VIDEO_FRMIVAL_TYPE_STEPWISE = 2,
```

Comment:
If `stepwise.min == stepwise.max` (and i.e. `stepwise.increment == 1` by convention), then `STEPWISE` becomes a `DISCRETE` value? This could be an opportunity to remove the distinction in the code and type definitions.

Reply:
I can't understand. If you remove STEP WISE, what will you describe the supported framerates if your driver supports all framerates from 1 to 1000 fps ? You will store 1000 values in an array ?

Reply:
I should have precised my thoughts better.
I thought about only keeping STEPWISE and using STEPWISE for DISCRETE values, rather than the other way around.

Reply:
Sorry, but I don't see we could save code space here with your proposal. You save one structure in video.h by removing the DISCRETE type but in N drivers code, you waste more code space because STEP_WISE need more fields compared to DISCRETE (e.g., min, max, step).

A part from that, if I have a sensor that supports only 15, 20, 60 fps, I would describe it with a DISCRETE type with 3 values which are simple and "natural" than saying that it supports 3 STEP_WISE ranges:

- the 1st range is from 15 to 15 with step = 1
- the 2nd range is from 20 to 20 with step = 1
- the 3rd range is from 60 to 60 with step = 1

Reply:
Sorry for the very delayed answer, I missed your reply.
I understand now: the overhead of dealing with a cumbersome API ("a to a" instead of "a") is greater than just 1 saved struct.
Thank you!

Reply:
@josuah : Sorry if I misunderstand your last phrase because I am not sure if you agree or disagree with my explanation :) ?
In my previous comment, I mean that if we remove the DISCRETE type, we will save one struct in video.h, but it will be not natural to describe the discrete framerate (with the above example). Moreover, I think we will waste some code space in drivers because the STEP_WISE struct has more fields than the DISCRETE struct if we use it to describe the discrete framerates.

Reply:
I do agree with your explanation! :)
I understand the benefit of introducing an extra struct for the sake of making the configuration of drivers less tedious and confusing, and waste code space!